oraclizeAPI_0.5.sol

```diff
@@ -77,760 +73,881 @@ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+
+    function oraclize_query(string memory _datasource, bytes[5] memory _args) oraclizeAPI internal returns (bytes32 id) {
         bytes[] memory dynargs = new bytes[](5);
-        dynargs[0] = args[0];
-        dynargs[1] = args[1];
-        dynargs[2] = args[2];
-        dynargs[3] = args[3];
-        dynargs[4] = args[4];
-        return oraclize_query(timestamp, datasource, dynargs, gaslimit);
-    }
-    function oraclize_query(string datasource, bytes[5] args, uint gaslimit) oraclizeAPI internal returns (bytes32 id) {
+        dynargs[0] = _args[0];
+        dynargs[1] = _args[1];
+        dynargs[2] = _args[2];
+        dynargs[3] = _args[3];
+        dynargs[4] = _args[4];
+        return oraclize_query(_datasource, dynargs);
+    }
+
+    function oraclize_query(uint _timestamp, string memory _datasource, bytes[5] memory _args) oraclizeAPI internal returns (bytes32 id) {
         bytes[] memory dynargs = new bytes[](5);
-        dynargs[0] = args[0];
-        dynargs[1] = args[1];
-        dynargs[2] = args[2];
-        dynargs[3] = args[3];
-        dynargs[4] = args[4];
-        return oraclize_query(datasource, dynargs, gaslimit);
+        dynargs[0] = _args[0];
+        dynargs[1] = _args[1];
+        dynargs[2] = _args[2];
+        dynargs[3] = _args[3];
+        dynargs[4] = _args[4];
+        return oraclize_query(_timestamp, _datasource, dynargs);
     }
 
-    function oraclize_cbAddress() oraclizeAPI internal returns (address){
-        return oraclize.cbAddress();
+    function oraclize_query(uint _timestamp, string memory _datasource, bytes[5] memory _args, uint _gasLimit) oraclizeAPI internal returns (bytes32 id) {
+        bytes[] memory dynargs = new bytes[](5);
+        dynargs[0] = _args[0];
+        dynargs[1] = _args[1];
+        dynargs[2] = _args[2];
+        dynargs[3] = _args[3];
+        dynargs[4] = _args[4];
+        return oraclize_query(_timestamp, _datasource, dynargs, _gasLimit);
     }
-    function oraclize_setProof(byte proofP) oraclizeAPI internal {
-        return oraclize.setProofType(proofP);
+
+    function oraclize_query(string memory _datasource, bytes[5] memory _args, uint _gasLimit) oraclizeAPI internal returns (bytes32 id) {
+        bytes[] memory dynargs = new bytes[](5);
+        dynargs[0] = _args[0];
+        dynargs[1] = _args[1];
+        dynargs[2] = _args[2];
+        dynargs[3] = _args[3];
+        dynargs[4] = _args[4];
+        return oraclize_query(_datasource, dynargs, _gasLimit);
     }
-    function oraclize_setCustomGasPrice(uint gasPrice) oraclizeAPI internal {
-        return oraclize.setCustomGasPrice(gasPrice);
+
+    function oraclize_setProof(byte _proofP) oraclizeAPI internal {
+        return oraclize.setProofType(_proofP);
     }
 
-    function oraclize_randomDS_getSessionPubKeyHash() oraclizeAPI internal returns (bytes32){
-        return oraclize.randomDS_getSessionPubKeyHash();
+
+    function oraclize_cbAddress() oraclizeAPI internal returns (address) {

```

Comment:
Set to view

Reply:
Done. Raises a new compiler warning re it not knowing for sure whether or not state mutability is affected, usual one. Do we:

__*a)*__ Leave as view since we know it's accurate, but give people new to the API more scary compiler warnings?

or

__*b)*__ Leave as is since it changes nothing but leaves fewer compiler warnings to API users?